src/controllers/dynakube/connectioninfo/communication_hosts.go

```diff
@@ -16,3 +22,56 @@ func GetOneAgentCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclie
 	}
 	return communicationHosts
 }
+
+func GetActiveGateEndpointsAsCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclient.CommunicationHost {
+	activegateEndpointsString := dynakube.Status.ActiveGate.ConnectionInfoStatus.Endpoints
+	if activegateEndpointsString == "" {
+		return []dtclient.CommunicationHost{}
+	}
+	return parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString)
+}
+
+func parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString string) []dtclient.CommunicationHost {
+	endpointStrings := strings.Split(activegateEndpointsString, ",")
+	endpointCommunicationHosts := make(map[dtclient.CommunicationHost]bool, len(endpointStrings))
+	for _, endpointString := range endpointStrings {
+		if endpoint, err := parseEndpointToCommunicationHost(endpointString); err == nil {
+			endpointCommunicationHosts[endpoint] = true
+		}
```

Comment:
you are secretly not handling the error here, and that is why its not working

`hostPart, portPart, err := net.SplitHostPort(host)` this call will fail if the endpoint has no port and cause an error

Reply:
yeah good catch ^

Reply:
@0sewa0  / @wepudt wdyt about such version?

Instead of parsing (as I suggested wrongly) , we can use `func (*URL) Port` and func (*URL) Hostname`

- Hostname returns u.Host, stripping any valid port number if present.
- Port returns the port part of u.Host, without the leading colon.

```
func parseEndpointToCommunicationHost(endpointString string) (dtclient.CommunicationHost, error) {
	if endpointString == "" {
		return dtclient.CommunicationHost{}, errors.New("empty endpoint string not allowed")
	}
	parsedEndpoint, err := url.Parse(endpointString)

	if err != nil {
		return dtclient.CommunicationHost{}, err
	}
	port := parsedEndpoint.Port()
	
	if port == "" {
		port = "80"
	}

	if err != nil {
		return dtclient.CommunicationHost{}, err
	}

	intPort, err := strconv.ParseUint(port, 10, 32)
	if err != nil {
		return dtclient.CommunicationHost{}, err
	}

	return dtclient.CommunicationHost{
		Protocol: parsedEndpoint.Scheme,
		Host:     parsedEndpoint.Hostname(),
		Port:     intPort,
	}, nil
}
```

Reply:
good input, thanks you two- submitted a solution including your suggestions